### PR TITLE
WIP #4516: Move formatPhoneNumber (and libphonenumber-js) server-side to reduce client bundle

### LINKED ISSUE #4516: Dynamically import libphonenumber-js
## Prerequisites
* [x] Are you running the latest version? 1.15.0
* [x] Are you able to consistently reproduce the issue?
* [x] Did you search the issue queue for existing issue?

## Work
The libphonenumber-js NPM module is currently included in the client bundle (122kb). It's only use is in `formatPhoneNumber` in lib/api/helpers.js. Dynamically import the library to reduce the client bundle size.

### PR DESCRIPTION
**Pull request by dancastellon**
Resolves #4516 
Impact: **minor**  
Type: **performance**

## Issue
libphonenumber-js was being included in the client bundle. It's only use is in `formatPhoneNumber`.

## Solution
Turn `formatPhoneNumber` into an async function that dynamically imports libphonenumber-js. In a separate PR, update the plugin that uses it (SMS in Reaction Contrib repo)

## Breaking changes
None

## Testing
1. Add the following code to /client/api/index.js:
import { formatPhoneNumber } from "/lib/api/helpers";
async function formatTest() {
  const num = await formatPhoneNumber("5555555555", "US");
  console.log("Formated phone number:", num);
}
formatTest();
2. Start the app and confirm you see in browser console: Formated phone number: 15555555555
3. Confirm bundle size is improved by running visualizer: `export TOOL_NODE_FLAGS='--max-old-space-size=4096' && METEOR_DISABLE_OPTIMISTIC_CACHING=1 meteor run --production --extra-packages bundle-visualizer`

## Before
libphone-before

## After
libphone-after

### CODE REVIEW

**Comment by zenweasel** [comment]
I don't think we should remove this. This is functionality we should provide. Would be nice if it could be dynamically imported though

**Comment by dancastellon** [comment]
@zenweasel Thanks, I've updated this to use a dynamic import instead of removing it altogether. @ajporlante @Akarshit - this is ready for review.

File: package.json [hunk 1]
```diff
@@ -76,7 +76,7 @@
     "immutable": "^3.8.2",
     "jquery-i18next": "^1.2.1",
     "later": "^1.2.0",
-    "libphonenumber-js": "^1.0.24",
+    "libphonenumber-js": "1.0.24",
```
[the remark this answers is not in the document]
Comment (dancastellon) [reply package.json]: @zenweasel Updated libphonenumber-js and retested.

**Comment by aldeed** [comment]
@dancastellon @zenweasel As far as I can tell, only SMS pkg is using this, and it uses it from server code. Going forward, the preferred way for clients to get a formatted phone number would be for it to be made available through GraphQL (e.g., a `displayPhone` field) already formatted by the server.

So given these facts, how about we move this function to `/imports/plugins/core/core/server/util`? That will prevent it from being included in the client bundle, so you could then keep it as a normal non-dynamic import.

If we want to keep SMS pkg working, you could update that one line that imports it in that repo.

**Comment by dancastellon** [comment]
Thanks @aldeed, PR updated

**Pull request merged**